### PR TITLE
ZooKeeper connection string 

### LINKED ISSUE #35: zk_connection_string should always expect a string...
As noted by @rfielding in https://github.com/DecipherNow/gm-fabric-go/issues/191 our behavior with parsing the `zk_connection_string` value is somewhat unexpected.  A ZooKeeper connection string is typically defined as a comma separated list in the form `ip[:port][,ip:[port]]...`.  Our use of viper.GetStringSlice actually does some conditional parsing that is not terribly clear.

In fact, it's more than just the fact that this is standard practice for defining ZooKeeper connection strings.  It is also misleading to have a configuration called `zk_connection_string` and expect people to set it to an array when they have multiple servers like so:
```
zk_connection_string = ["10.0.0.1:2181", "10.0.0.2:2181", "10.0.0.3:2181"]
```
We should only be accepting a string in the standard format.

### PR DESCRIPTION
**Pull request by dougfort**
is typically defined as a comma separaed list in the form ip[:port][,ip:[port]]....

closes #35

### CODE REVIEW

File: default/templates/cmd/server/zk.go [hunk 1]
```diff
@@ -81,6 +83,10 @@ func notifyZkOfGatewayEndpointIfNeeded(logger zerolog.Logger) ([]zkCancelFunc, e
 	return []zkCancelFunc{cancel}, nil
 }
 
+func parseConnectionString() []string {
```
Comment (joshua-rutherford) [inline default/templates/cmd/server/zk.go]: Why not just extract the connection string as a variable up top and reuse it?  Seems cleaner to do:
```go
zk_servers := strings.Split(viper.GetString("zk_connection_string"), ",")
```

Comment (joshua-rutherford) [reply default/templates/cmd/server/zk.go]: I guess my point is, you've extracted a one liner into a function without parameters which adds unneeded indirection.

Comment (dougfort) [reply default/templates/cmd/server/zk.go]: Reuse? It's in three different functions. You mean as a module level variable? I don't  think you can do that. And if you can, you shouldn't.

I suspect that there shouldn't be three different functions, but we should have a separate issue for that.

**Comment by joshua-rutherford** [comment]
Fair point. I wasn't looking at the whole file, just the changes so I did not realize these were in separate files. I'm sure this would be resolved by removing the code duplication.

**Pull request merged**